src/Collection/Collection.router.ts

```diff
@@ -380,7 +380,7 @@ export class CollectionRouter extends Router {
     let upsertedCollection: CollectionAttributes
 
     try {
-      if (isTPCollectionURN(collectionJSON.urn)) {
+      if (collectionJSON.urn && isTPCollectionURN(collectionJSON.urn)) {
```

Comment:
What do you think about moving the undefined check for `collectionJSON.urn` into `isTPCollectionURN`? It's a condition to have a non-undefined or non-nulled urn to be a TP collection, so the check could live there and be used elsewhere too.